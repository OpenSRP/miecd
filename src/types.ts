import { SmsData, CompartmentSmsTypes } from './store/ducks/sms_events';

// typings
export type SmsFilterFunction = (smsData: SmsData) => boolean;
export type CompartmentsSmsFilterFunction = (CompartmentSmsData: CompartmentSmsTypes) => boolean;
