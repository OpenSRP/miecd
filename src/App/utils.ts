import { DEFAULT_PLAN_ID, KEYCLOAK_API_BASE_URL, KEYCLOAK_USERS_PAGE_SIZE, OPENSRP_API_BASE_URL } from 'configs/env';
import { URL_LOCATION_UNIT } from '../constants';

const openSrpRestBase = OPENSRP_API_BASE_URL + '/rest/';

export const baseProps = {
    baseURL: openSrpRestBase,
    opensrpBaseURL: openSrpRestBase,
    keycloakBaseURL: KEYCLOAK_API_BASE_URL,
};

export const locationUnitProps = {
    ...baseProps,
    filterByParentId: false,
};

export const newLocationUnitProps = {
    ...locationUnitProps,
    successURLGenerator: () => URL_LOCATION_UNIT,
    cancelURLGenerator: () => URL_LOCATION_UNIT,
    hidden: ['serviceType', 'latitude', 'longitude'],
};

export const editLocationProps = {
    ...newLocationUnitProps,
    ...locationUnitProps,
};

export const usersListProps = {
    ...baseProps,
    usersPageSize: KEYCLOAK_USERS_PAGE_SIZE,
};

export const teamAssignmentProps = {
    ...baseProps,
    defaultPlanId: DEFAULT_PLAN_ID,
};

export const createEditUserProps = {
    ...baseProps,
    userFormHidden: [],
};
