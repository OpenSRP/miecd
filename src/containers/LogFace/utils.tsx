import { replace, split, trim } from 'lodash';
import React from 'react';

/** convert the smsData message field from prose to more easily readable
 * point format
 *
 * @param message - the message prose.
 */
export const parseMessage = (message: string) => {
    const propValues = split(message, '\n');
    const replacedEquals = propValues.map(trim).map((entry) => replace(entry, / =\s*/, ' : '));
    return (
        <ul>
            {replacedEquals.map((value, index) => {
                return <li key={`${value}-${index}`}>{value}</li>;
            })}
        </ul>
    );
};
