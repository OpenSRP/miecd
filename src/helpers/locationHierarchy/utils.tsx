import { ParsedHierarchyNode, RawHierarchyNode, RawHierarchyNodeMap, RawOpenSRPHierarchy, TreeNode } from './types';
import { cloneDeep } from 'lodash';
import cycle from 'cycle';
import TreeModel from 'tree-model';

/** Parse the raw child hierarchy node map
 *
 * @param {RawHierarchyNodeMap} rawNodeMap - Object of raw hierarchy nodes
 * @param {string} parent - node parent id
 * @returns {Array<ParsedHierarchyNode>} Array of Parsed hierarchy nodes
 */
const parseChildren = (rawNodeMap: RawHierarchyNodeMap, parent: string) => {
    // skip key when destructuring
    const rawHierarchyNode: RawHierarchyNode[] = Object.entries(rawNodeMap).map(([, value]) => value);
    return rawHierarchyNode.map((child) => {
        const parsedNode: ParsedHierarchyNode = {
            ...child,
            title: child.label,
            key: `${child.id}-${parent}`,
            children: child.children ? parseChildren(child.children, parent) : [],
        };
        return parsedNode;
    });
};

/** parses the raw opensrp hierarchy to a hierarchy that we can quickly build
 * our tree model from.
 *
 * @param {RawOpenSRPHierarchy} raw - the response we get from opensrp
 * @returns {ParsedHierarchyNode} - returns Parent node with its children
 */
const parseHierarchy = (raw: RawOpenSRPHierarchy) => {
    // clone the locationTree, we are going to be mutating a copy
    const rawClone: RawOpenSRPHierarchy = cloneDeep(raw);

    // !IMPORTANT ASSUMPTION : locationsTreeClone has a single object under map, i.e there is only one root jurisdiction
    const { map } = rawClone.locationsHierarchy;
    // !IMPORTANT ASSUMPTION : locationsTreeClone has a single object under map, i.e there is only one root jurisdiction
    // skip key when destructuring
    const rawNode: RawHierarchyNode = Object.entries(map).map(([, value]) => value)[0];
    const parsedNode: ParsedHierarchyNode = {
        ...rawNode,
        title: rawNode.label,
        key: rawNode.id,
        children: rawNode.children ? parseChildren(rawNode.children, rawNode.id) : [],
    };

    return parsedNode;
};

/** takes the raw opensrp hierarchy response and creates a tree model structure
 *
 * @param {RawOpenSRPHierarchy} apiResponse - the response we get from opensrp
 * @returns {TreeNode} - returns root node
 */
export const generateJurisdictionTree = (apiResponse: RawOpenSRPHierarchy): TreeNode => {
    const tree = new TreeModel();
    const hierarchy = parseHierarchy(apiResponse);
    const root = tree.parse<ParsedHierarchyNode>(hierarchy);
    return root;
};

/**
 * serialize tree due to circular dependencies
 *
 * @param trees - trees to be serialized
 */
export const serializeTree = (trees: (TreeNode | undefined)[]) => {
    return JSON.stringify(trees.map((tree) => JSON.stringify(cycle.decycle(tree))));
};
