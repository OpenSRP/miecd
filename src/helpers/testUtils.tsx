import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
