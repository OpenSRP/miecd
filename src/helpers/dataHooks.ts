import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
