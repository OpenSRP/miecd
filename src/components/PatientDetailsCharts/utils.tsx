import NoRecord from 'components/NoRecord';
import Ripple from 'components/page/Loading';
import { TFunction, useTranslation } from 'react-i18next';
import supersetFetch from 'services/superset';
import { getMonthNames } from '../../constants';
import { sortByEventDate } from '../../helpers/utils';
import { ChildChartData, MotherChartData } from 'store/ducks/chartData';
import superset, { SupersetFormData } from '@onaio/superset-connector';
import { ActionCreator } from 'redux';
import { GET_FORM_DATA_ROW_LIMIT } from 'configs/env';
import React from 'react';

/** factory for form data filters for chart pages */
export const chartFormDataFilters = (patientId: string) =>
    superset.getFormData(GET_FORM_DATA_ROW_LIMIT, [
        {
            comparator: patientId,
            operator: '==',
            subject: 'patient_id',
        },
    ]);

/** abstracts code that actually makes the superset Call since it is quite similar
 * @param supersetSlice - slice string
 * @param actionCreator - redux action creator
 * @param supersetService - the supersetService
 * @param supersetOptions - adhoc filters for superset call
 */
export async function chartSupersetCall<TAction, TResponse>(
    supersetSlice: string,
    actionCreator: ActionCreator<TAction>,
    supersetService: typeof supersetFetch = supersetFetch,
    supersetOptions: SupersetFormData | null = null,
) {
    const asyncOperation = supersetOptions
        ? supersetService(supersetSlice, supersetOptions)
        : supersetService(supersetSlice);
    return asyncOperation
        .then((result: TResponse[] | undefined) => {
            if (result) {
                actionCreator(result);
            }
        })
        .catch((error) => {
            throw error;
        });
}

/** props for BrokenLoading chart */
export interface BrokenLoadingChartPageProps {
    error?: Error;
    loading: boolean;
    broken: boolean;
}

/** dry component for showing loader or broken page */
export const BrokenLoadingChartPage = ({ error, loading, broken }: BrokenLoadingChartPageProps) => {
    const { t } = useTranslation();
    if (loading) {
        return <Ripple></Ripple>;
    }

    if (broken) {
        return <NoRecord message={t(`Chart could not load, an error occurred. ${error?.message ?? ''}`)} />;
    }
    return null;
};

/** create a chart data series containing the weight and height data series from smsEvents */
export const getWeightHeightDataSeries = (smsEvents: ChildChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'weight' },
        { data: [], name: 'height' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        if (data.weight && data.height) {
            categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
            dataSeries[0].data.push(data.weight);
            dataSeries[1].data.push(data.height);
        }
    }
    return {
        categories,
        dataSeries,
    };
};

/** create a chart data series containing the weight data series from smsEvents */
export const getWeightDataSeries = (smsEvents: MotherChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [{ data: [], name: 'weight' }];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        if (data.weight) {
            categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
            dataSeries[0].data.push(data.weight);
        }
    }
    return {
        categories,
        dataSeries,
    };
};

/** create a data series containing the diastolic and systolic data series from smsEvents */
export const getBloodPSeriesForChart = (smsEvents: MotherChartData[], t: TFunction) => {
    const categories: string[] = [];
    const dataSeries: { data: (number | undefined)[]; name: string }[] = [
        { data: [], name: 'systolic' },
        { data: [], name: 'diastolic' },
    ];
    // make sure events are sorted from oldest to latest
    const sortedSmsEvents = sortByEventDate(smsEvents).reverse();

    const regex = /\d{2,3}\S{0,3}\d{2,3}/;
    const singleBpValueRegex = /\d{2,3}/g;

    for (const data of sortedSmsEvents) {
        const calendarCategoryObj = {
            month: new Date(data.event_date).getMonth(),
            year: new Date(data.event_date).getFullYear(),
        };
        const hasValidBp = regex.test(data.bp);
        if (!hasValidBp) {
            continue;
        }
        categories.push(`${getMonthNames(t)[calendarCategoryObj.month]}/${calendarCategoryObj.year}`);
        const matchedValues = [...Array.from(data.bp.matchAll(singleBpValueRegex))];

        dataSeries[0].data.push(Number(matchedValues[0]));
        dataSeries[1].data.push(Number(matchedValues[1]));
    }
    return {
        categories,
        dataSeries,
    };
};
