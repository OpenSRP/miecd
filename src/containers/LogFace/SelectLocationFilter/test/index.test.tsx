import { Dictionary } from '@onaio/utils/dist/types/types';
import toJson from 'enzyme-to-json';
import { mountWithTranslations } from 'helpers/testUtils';
import React from 'react';
import { RawOpenSRPHierarchy } from 'store/ducks/locationHierarchy/types';
import { generateJurisdictionTree } from 'store/ducks/locationHierarchy/utils';
import * as securityAuthenticate from 'store/ducks/tests/fixtures/securityAuthenticate.json';
import { SelectLocationFilter } from '..';

jest.mock('react-select', () => {
    const Select = ({ options, onChange, onInputChange }: Dictionary) => {
        function handleChange(event: Dictionary) {
            const option = options.find((option: Dictionary) => option.value === event.target.value);
            onChange(option);
        }
        function handleInputChange(event: Dictionary) {
            onInputChange(event.target.value);
        }

        return (
            <select data-testid="select" onChange={handleChange} onInput={handleInputChange}>
                {options.map(({ label, value }: Dictionary) => (
                    <option key={value} value={value}>
                        {label}
                    </option>
                ))}
            </select>
        );
    };
    return Select;
});

describe('selectLocationFilter', () => {
    const userLocationTree = generateJurisdictionTree(securityAuthenticate.locations as unknown as RawOpenSRPHierarchy);
    const demoUserLocationId = 'd1865325-11e6-4e39-817b-e676c1affecf';

    it('works correctly', () => {
        const locationChangeMock = jest.fn();
        const props = {
            userLocationTree,
            userLocationId: demoUserLocationId,
            onLocationChange: locationChangeMock,
        };
        const wrapper = mountWithTranslations(<SelectLocationFilter {...props} />);

        // list option; should be 3, demo user is at national level and has access to only 3 provinces
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('initial Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeFalsy();

        // simulate selection of one of provinces
        wrapper
            .find('select')
            .simulate('change', { target: { value: 'eccfe905-0e03-4188-98bc-22f141cccd0e', label: 'Kon Tum' } });

        // select now has the next set of options and menu is still open
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('district Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeFalsy();

        // simulate selection of one of districts
        wrapper
            .find('select')
            .simulate('change', { target: { value: 'e1488a4b-d11f-463c-b20c-56d4fc190ab3', label: 'Đaklei' } });

        // select now has the next set of options and menu is still open
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('commune Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeFalsy();

        // simulate selection of one of communes
        wrapper
            .find('select')
            .simulate('change', { target: { value: 'd2c63216-ac3c-499c-9688-94ede9308275', label: 'Đăk Pek' } });

        // select now has the next set of options and menu is still open
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('village Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeFalsy();

        // simulate selection of one of communes
        wrapper
            .find('select')
            .simulate('change', { target: { value: '020d8cfd-a843-4405-956a-24a778806bc9', label: 'Thôn 14 A' } });

        // select now has the next set of options and menu is now closed on select
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('still showing village Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeTruthy();

        expect(locationChangeMock).toHaveBeenCalledTimes(4);
        expect(locationChangeMock.mock.calls).toEqual([
            ['eccfe905-0e03-4188-98bc-22f141cccd0e'],
            ['e1488a4b-d11f-463c-b20c-56d4fc190ab3'],
            ['d2c63216-ac3c-499c-9688-94ede9308275'],
            ['020d8cfd-a843-4405-956a-24a778806bc9'],
        ]);
        wrapper.unmount();
    });

    it('test input filter', () => {
        // simulate searching behavior
        const props = {
            userLocationTree,
            userLocationId: demoUserLocationId,
            onLocationChange: () => void 0,
        };
        const wrapper = mountWithTranslations(<SelectLocationFilter {...props} />);

        // list option; should be 3, demo user is at national level and has access to only 3 provinces
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('initial Options');
        expect((wrapper.find('Select').props() as Dictionary).closeMenuOnSelect).toBeFalsy();

        // expect(wrapper.find('Select').props()).toEqual();

        // target village 'Thôn 14 A'
        wrapper.find('select').simulate('input', {
            target: {
                value: 'Thôn 14',
            },
        });

        // check options
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('options shown during search');

        // see what clearing does
        wrapper.find('select').simulate('input', {
            target: {
                value: '',
            },
        });

        // should show initial options
        expect(toJson(wrapper.find('select'))).toMatchSnapshot('and back to initial options');

        wrapper.unmount();
    });
});
