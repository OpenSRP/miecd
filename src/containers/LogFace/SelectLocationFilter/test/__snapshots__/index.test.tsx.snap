// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectLocationFilter test input filter: and back to initial options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
    value="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
  >
    Gia Lai
  </option>
  <option
    key="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
    value="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
  >
    Điện Biên
  </option>
  <option
    key="eccfe905-0e03-4188-98bc-22f141cccd0e"
    value="eccfe905-0e03-4188-98bc-22f141cccd0e"
  >
    Kon Tum
  </option>
</select>
`;

exports[`selectLocationFilter test input filter: initial Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
    value="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
  >
    Gia Lai
  </option>
  <option
    key="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
    value="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
  >
    Điện Biên
  </option>
  <option
    key="eccfe905-0e03-4188-98bc-22f141cccd0e"
    value="eccfe905-0e03-4188-98bc-22f141cccd0e"
  >
    Kon Tum
  </option>
</select>
`;

exports[`selectLocationFilter test input filter: options shown during search 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="020d8cfd-a843-4405-956a-24a778806bc9"
    value="020d8cfd-a843-4405-956a-24a778806bc9"
  >
    Vietnam &gt; Kon Tum &gt; Đaklei &gt; Đăk Pek &gt; Thôn 14 A
  </option>
  <option
    key="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
    value="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
  >
    Vietnam &gt; Kon Tum &gt; Đaklei &gt; Đăk Pek &gt; Thôn 14 B
  </option>
</select>
`;

exports[`selectLocationFilter works correctly: commune Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="d114e2f4-d906-483c-b968-20557b19b738"
    value="d114e2f4-d906-483c-b968-20557b19b738"
  >
    Thị trấn Đăklei
  </option>
  <option
    key="d2c63216-ac3c-499c-9688-94ede9308275"
    value="d2c63216-ac3c-499c-9688-94ede9308275"
  >
    Đăk Pek
  </option>
  <option
    key="6b2265c5-6d34-4857-aa0d-69ea8fcfce85"
    value="6b2265c5-6d34-4857-aa0d-69ea8fcfce85"
  >
    Đăk Choong
  </option>
</select>
`;

exports[`selectLocationFilter works correctly: district Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="66e45e03-be29-4a18-99d9-cb625347f2d9"
    value="66e45e03-be29-4a18-99d9-cb625347f2d9"
  >
    Tu Mo Rông
  </option>
  <option
    key="caf90e84-4d2f-42f5-b96c-081107b40558"
    value="caf90e84-4d2f-42f5-b96c-081107b40558"
  >
    Kon Rẫy
  </option>
  <option
    key="e1488a4b-d11f-463c-b20c-56d4fc190ab3"
    value="e1488a4b-d11f-463c-b20c-56d4fc190ab3"
  >
    Đaklei
  </option>
</select>
`;

exports[`selectLocationFilter works correctly: initial Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
    value="c8054fa5-2d63-4cb5-bc52-173a4eb367fd"
  >
    Gia Lai
  </option>
  <option
    key="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
    value="1b8b7a76-15e1-446b-9d1e-c8cc97777a28"
  >
    Điện Biên
  </option>
  <option
    key="eccfe905-0e03-4188-98bc-22f141cccd0e"
    value="eccfe905-0e03-4188-98bc-22f141cccd0e"
  >
    Kon Tum
  </option>
</select>
`;

exports[`selectLocationFilter works correctly: still showing village Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="28ebc596-bbf2-4c34-b879-6f9f5eff5b83"
    value="28ebc596-bbf2-4c34-b879-6f9f5eff5b83"
  >
    Thôn Đăk Dền
  </option>
  <option
    key="020d8cfd-a843-4405-956a-24a778806bc9"
    value="020d8cfd-a843-4405-956a-24a778806bc9"
  >
    Thôn 14 A
  </option>
  <option
    key="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
    value="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
  >
    Thôn 14 B
  </option>
  <option
    key="5649d82b-5357-4b7b-bdb6-618805cc68d3"
    value="5649d82b-5357-4b7b-bdb6-618805cc68d3"
  >
    Thôn Đăk Trấp
  </option>
</select>
`;

exports[`selectLocationFilter works correctly: village Options 1`] = `
<select
  data-testid="select"
  onChange={[Function]}
  onInput={[Function]}
>
  <option
    key="28ebc596-bbf2-4c34-b879-6f9f5eff5b83"
    value="28ebc596-bbf2-4c34-b879-6f9f5eff5b83"
  >
    Thôn Đăk Dền
  </option>
  <option
    key="020d8cfd-a843-4405-956a-24a778806bc9"
    value="020d8cfd-a843-4405-956a-24a778806bc9"
  >
    Thôn 14 A
  </option>
  <option
    key="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
    value="22cdc2ee-2137-47a7-8b85-9cd194b8bdfe"
  >
    Thôn 14 B
  </option>
  <option
    key="5649d82b-5357-4b7b-bdb6-618805cc68d3"
    value="5649d82b-5357-4b7b-bdb6-618805cc68d3"
  >
    Thôn Đăk Trấp
  </option>
</select>
`;
